src/glfs-flock.c

```diff
@@ -0,0 +1,351 @@
+/**
+ * A utility to request for full file advisory lock on a file from remote Gluster volume.
+ *
+ * Copyright (C) 2016 Red Hat Inc.
+ */
+
+#include <config.h>
+
+#include "glfs-flock.h"
+#include "glfs-util.h"
+
+#include <errno.h>
+#include <error.h>
+#include <getopt.h>
+#include <glusterfs/api/glfs.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define AUTHORS "Written by Anoop C S."
+
+/**
+ * Used to store the state of the program, including user supplied options.
+ *
+ * gluster_url  : Struct of the parsed url supplied by the user.
+ * xlator_option: Stores various xlator options supplied by user.
+ * url          : Full url used to find the remote file or directory (supplied by user).
+ * debug        : Whether to log additional debug information.
+ * l_type       : Stores the requested lock type.
+ * block        : Whether to block the lock request in case of a conflicting lock.
+ */
+struct state {
+        struct gluster_url *gluster_url;
+        struct xlator_option *xlator_options;
+        char *url;
+        bool debug;
+        bool block;
+        short l_type;
+};
+
+static struct state *state;
+
+static struct option const long_options[] =
+{
+        {"debug", no_argument, NULL, 'd'},
+        {"exclusive", no_argument, NULL, 'e'},
+        {"help", no_argument, NULL, 'h'},
+        {"nonblock", no_argument, NULL, 'n'},
+        {"xlator-option", required_argument, NULL, 'o'},
+        {"port", required_argument, NULL, 'p'},
+        {"shared", no_argument, NULL, 's'},
+        {"unlock", no_argument, NULL, 'u'},
+        {"version", no_argument, NULL, 'v'},
+        {NULL, no_argument, NULL, 0}
+};
+
+static void
+usage ()
+{
+        printf ("Usage: %s [OPTION]... URL\n"
+                "Request a full file advisory lock on files from a remote Gluster volume.\n\n"
```

Comment:
Since it seems that this utility can only be executed when the shell has an active connection, some of these options don't make much sense (I'm assuming this is just from using another command as a template).

Namely, the xlator-option and port flags. Correct me if I'm wrong, however.